shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -65,4 +75,106 @@ public static MessageHeader findOrInitializeMessageHeader(Bundle bundle) {
         }
         return (MessageHeader) messageHeader;
     }
+
+    /**
+     * Updates the value of an identifier for an Organization within a FHIR Bundle based on specific
+     * criteria. This method processes each Patient resource within the bundle, identifies linked
+     * Organizations via the assigner field of Patient identifiers, and checks each Organization's
+     * identifier for specific extension criteria. If the criteria are met (specifically an
+     * extension URL and a valueString that match predefined values), the identifier's value is
+     * updated to a new specified value.
+     *
+     * <p>The following HL7 segment is changed: PID 3.4 - Assigning Authority
+     *
+     * @param bundle the FHIR Bundle containing Patient resources whose identifiers need to be
+     *     updated.
+     * @param newValue the new value to which the identifier type code should be set.
+     */
+    public static void updateOrganizationIdentifierValue(Bundle bundle, String newValue) {
```

Comment:
I think I will end up moving this logic into the transform method of the transformation class. This will only leave helper methods in the HapiHelper class.

Reply:
I will use FHIR path for checking, if it exists, then I will proceed to update the value